File: Java/Language/Runtime/ReflectionTest2/src/HomeLoan.java
package finance;

public class HomeLoan {

	@MaxDuration(12)
	public float common(double amount, int period) {
		return amount < 1000000 ? 9 : 8;
	}

	public float woman(double amount, int period) {
		return common(amount, period) - 1;
	}

	public float soldier(double amount, int period) {
		return 4;
	}
}

File: dotnet/Language/Runtime/ReflectionTest2/BankLib/HomeLoan.cs
namespace Finance;

public class HomeLoan
{
    public float Common(double amount, int period) => amount < 1000000 ? 8.5f : 8.0f;

    [MaxDuration(Limit=12)]
    public float Woman(double amount, int period) => Common(amount, period) - 1;

    public float Soldier(double amount, int period) => 4.0f;
}
